src/chains/ethereum/ethereum/tests/forking/forking.test.ts

```diff
@@ -900,20 +923,30 @@ describe("forking", function () {
               const subId = await remoteProvider.send("eth_subscribe", [
                 "newHeads"
               ]);
-              // set the remoteProvider's value1 initialValue to {remoteInitialValue}
-              await set(remoteProvider, 1, remoteInitialValue);
-              const message = await remoteProvider.once("message");
-              await remoteProvider.send("eth_unsubscribe", [subId]);
-              const blockNumber = parseInt(
-                (message.data.result as any).number,
-                16
+              // set the remoteProvider's value1 initialValue to {remoteInitialValue} (only if not null)
+              if (remoteInitialValue !== null) {
+                await set(remoteProvider, 1, remoteInitialValue);
+                await remoteProvider.once("message");
+                await remoteProvider.send("eth_unsubscribe", [subId]);
+                const blockNumber = await getBlockNumber(remoteProvider);
+                assert.strictEqual(
+                  parseInt(
+                    await get(remoteProvider, "value1", blockNumber),
+                    16
+                  ),
+                  remoteInitialValue
+                ); // sanity check to make sure our initial conditions are correct
+              }
+
+              const blockNumber = await getBlockNumber(remoteProvider);
+              const startValue = await get(
+                remoteProvider,
+                "value1",
+                +blockNumber
```

Comment:
```suggestion
                blockNumber
```
`getBlockNumber` already returns a `number`